pixeldata/src/lib.rs

```diff
@@ -2155,12 +2155,42 @@ where
             .fail()?;
         }
 
-        let rescale = zip(&rescale_intercept, &rescale_slope)
+        let rescale_data = zip(&rescale_intercept, &rescale_slope)
             .map(|(intercept, slope)| Rescale {
                 intercept: *intercept,
                 slope: *slope,
             })
-            .collect();
+            .collect::<Vec<Rescale>>();
+
+        let rescale = {
+            if rescale_data.len() > frame as usize {
+                vec![rescale_data[frame as usize]]
+            } else if !rescale_data.is_empty() {
+                vec![rescale_data[0]]
```

Comment:
Maybe even better:
```suggestion
            } else if let Some(resc) = rescale_data.get(0) {
                vec![*resc]
```
The same trick can be done in the other cases.

Reply:
Switched the code from the if blocks to a functional style.